controllers/providercredential/provider-credential-controller.go

```diff
@@ -80,15 +80,15 @@ func (r *ProviderCredentialSecretReconciler) Reconcile(ctx context.Context, req
 		}
 	}
 
-	//We need to extract the specific secret.Data
-	secretData, err := extractImportantData(secret)
+	//validate provider type label
+	_, err := extractImportantData(secret)
```

Comment:
The extractImportantData was done because the provider secret and the copy of that secret that it was often connected two did NOT contain all the same information.

ie. The AWS secret created by the UI from the Cloud Provider secret, does not contain all the OTHER cloud provider data, like the basedomain, etc...  I guess it doesn't do any damage, to store the extra information, but it also increases the trigger possibilities.  

if you change the basedomain, your now going to massively increase the secret size, and your storing things like SSH keys and basedomain with a cloud credential secret, that were not there before nor required by Hive.

I would say, you are better off just adding the additional field if it is part of the Hive cloud provider secret, and a NEW test.

Reply:
clarified by Josh, the copied secret from UI only has the important data. which doesn't contain all provider data. We are good to revert the code change